include/aws/http/private/connection_impl.h

```diff
@@ -42,9 +42,9 @@ struct aws_http_connection_vtable {
     void (*close)(struct aws_http_connection *connection);
     bool (*is_open)(const struct aws_http_connection *connection);
     bool (*new_requests_allowed)(const struct aws_http_connection *connection);
-    void (*update_window)(struct aws_http_connection *connection, size_t increment_size);
 
     /* HTTP/2 specific functions */
+    int (*update_window)(struct aws_http_connection *connection, size_t increment_size);
```

Comment:
from `void` -> `int`. Also start to think about maybe it's better to have a return code for stream window update as well? It can fail for h2 as well. (Will never fail for h1)